### ISSUE TITLE
[BUG] Multirocket first order difference features miscalculations

### ISSUE THREAD

**Issue by leandrepi** [sktime/sktime#11291]
**Describe the bug**
While writing a cython version of the `MultiRocket`estimator, I found two apparent bugs in the current computation of the first-order difference features, also present in the [original implementation](https://github.com/ChangWeiTan/MultiRocketl). 

The most major of the two bugs only affects the multivariate version: there, the `combination_index` is never updated in the `X1` feature computation loop, which means **every kernel in the differenced pass reuses combination 0's channel selection**.

The other bug affects both the univariate and multivariate versions, but is rather minor. It's an oversight of the fact that `X1 = np.diff(X, 1)` has one less sample than X in its last dimension; the code computes `end` from `X`'s shape in both feature computations loops, thereby creating a slight mismatch in padding between `X` and `X1` features. 

**To Reproduce**

```python
"""Run: python multirocket_bug_repro.py"""

import numpy as np

from sktime.transformations.rocket import MultiRocket, MultiRocketMultivariate

KW = dict(num_kernels=84, max_dilations_per_kernel=32, random_state=0)


def bug1_channels_ignored():
    """Half the features are blind to most channels.

    Fit once, then overwrite one channel at a time with fresh noise and see
    what moves. Channels outside combination 0's selection change nothing in
    the differenced half.
    """
    print("=" * 72)
    print("BUG 1: differenced half ignores per-kernel channel selection")
    print("=" * 72)
    rng = np.random.RandomState(0)
    X = rng.normal(size=(4, 4, 60))
    est = MultiRocketMultivariate(**KW).fit(X)
    f0 = est.transform(X).to_numpy()
    nf = f0.shape[1] // 8

    comb0 = sorted(set(est.parameter[1][: est.parameter[0][0]].tolist()))
    print(f"channels used by combination 0: {comb0}\n")
    print("  channel   raw-half   differenced-half   (% of features that change)")
    for c in range(X.shape[1]):
        X2 = X.copy()
        X2[:, c] = rng.normal(size=(X.shape[0], X.shape[2]))
        f = est.transform(X2).to_numpy()
        raw = (~np.isclose(f0[:, : 4 * nf], f[:, : 4 * nf])).mean()
        dif = (~np.isclose(f0[:, 4 * nf :], f[:, 4 * nf :])).mean()
        print(f"     {c}       {raw:6.1%}          {dif:6.1%}")
    print("\nexpected: every channel influences both halves")


def bug2_window_misaligned():
    """Show diff(X) is not transformed the way the raw pass transforms X.

    Univariate, so bug 2 cannot contribute. The two estimators end up with
    bit-identical fitted parameters, so any difference is in the transform.
    """
    print()
    print("=" * 72)
    print("BUG 2: differenced half computes `end` from the raw series length")
    print("=" * 72)
    for cls in (MultiRocketMultivariate, MultiRocket):
        X = np.random.RandomState(0).normal(size=(4, 1, 60))
        Y = np.diff(X, 1)

        a = cls(**KW).fit(X)  # differenced half runs on diff(X)
        b = cls(**KW).fit(Y)  # raw half runs on the same series
        same_params = all(
            np.array_equal(u, v) for u, v in zip(a.parameter1, b.parameter)
        )

        fa = a.transform(X).to_numpy()
        fb = b.transform(Y).to_numpy()
        nf = fa.shape[1] // 8
        differing = (
            ~np.isclose(fa[:, 4 * nf :], fb[:, : 4 * nf], rtol=1e-4, atol=1e-5)
        ).mean()

        print(f"{cls.__name__}:")
        print(f"   fitted parameters identical : {same_params}")
        print(f"   yet features differing      : {differing:.1%}")
    print("\nexpected: 0% - same kernels, same biases, same series")


if __name__ == "__main__":
    bug1_channels_ignored()
    bug2_window_misaligned()
```

**Expected behavior**

For the first bug, you'd expect every channel to influence the resulting first-order difference features; for the second bug, the first-order difference features obtained for `X` should match the raw features we'd get by using a new instance on `np.diff(X)`.

**Additional context**

I intend to submit a small PR addressing those two issues shortly, as they hinder a port of this estimator to cython.

**Versions**
<details>

System:
    python: 3.12.11 (main, Aug 18 2025, 19:19:11) [Clang 20.1.4 ]
executable: /home/leandre/prog/misc/sktime/.venv/bin/python
   machine: Linux-6.12.101+deb13-amd64-x86_64-with-glibc2.41

Python dependencies:
          pip: None
       sktime: 1.1.0
 scikit-learn: 1.9.1
  scikit-base: 1.1.1
        numpy: 2.4.6
        scipy: 1.18.1
       pandas: 3.0.5
   matplotlib: None
       joblib: 1.6.0
huggingface-hub: None
        numba: 0.67.0
     pmdarima: None
pytorch-forecasting: None
   skforecast: None
        skpro: None
statsforecast: None
  statsmodels: None
 transformers: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None

</details>

**Comment by fkiraly** [comment]
huh, looks indeed like a bug.

FYI, if you later want to contribute a cython version, see here: https://github.com/sktime/sktime-cython/pulls - I think there are already cython versions of some of the rockets there.
